ako-infra/proxy/management_service_controller.go

```diff
@@ -0,0 +1,575 @@
+		payload,
+		&response,
+	)
+	if err != nil {
+		return fmt.Errorf("management service creation API call failed: %v", err)
+	}
+	utils.AviLog.Infof("VKS Management Service %s created successfully. Response: %v", c.serviceName, response)
+	return nil
+}
+
+func (c *ManagementServiceController) GetManagementService() (map[string]interface{}, error) {
+	dynamicClient := lib.GetDynamicClientSet()
+	if dynamicClient == nil {
+		return nil, fmt.Errorf("dynamic client not available")
+	}
+
+	gvr := schema.GroupVersionResource{
+		Group:    "netoperator.vmware.com",
+		Version:  "v1alpha1",
+		Resource: "managementservices",
+	}
+
+	resource, err := dynamicClient.Resource(gvr).Get(context.TODO(), c.serviceName, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil, fmt.Errorf("management service not found")
+		}
+		return nil, fmt.Errorf("failed to get management service: %v", err)
+	}
+
+	return resource.Object, nil
+}
+
+func (c *ManagementServiceController) validateManagementServiceConfig(serviceObj map[string]interface{}) bool {
+	spec, ok := serviceObj["spec"].(map[string]interface{})
+	if !ok {
+		utils.AviLog.Infof("ManagementService spec not found or invalid format")
+		return false
+	}
+
+	managementAddresses, ok := spec["managementAddresses"].([]interface{})
+	if !ok || len(managementAddresses) == 0 {
+		utils.AviLog.Infof("ManagementService managementAddresses not found or empty")
+		return false
+	}
+
+	expectedAddress := lib.GetControllerIP()
+	addressFound := false
+	for _, addr := range managementAddresses {
+		if addrStr, ok := addr.(string); ok && addrStr == expectedAddress {
+			addressFound = true
+			break
+		}
+	}
+	if !addressFound {
+		utils.AviLog.Infof("ManagementService expected address %s not found in managementAddresses", expectedAddress)
+		return false
+	}
+
+	ports, ok := spec["ports"].([]interface{})
+	if !ok || len(ports) == 0 {
+		utils.AviLog.Infof("ManagementService ports not found or empty")
+		return false
+	}
+
+	portFound := false
+	for _, portInterface := range ports {
+		if port, ok := portInterface.(map[string]interface{}); ok {
+			if value, ok := port["value"].(float64); ok && int32(value) == c.servicePort {
+				if tls, ok := port["tls"].(map[string]interface{}); ok {
+					if hostname, ok := tls["hostname"].(string); ok && hostname == expectedAddress {
+						if caCert, ok := tls["certificateAuthorityChain"].(string); ok && len(caCert) > 0 {
+							expectedCaCert := utils.SharedCtrlProp().GetAllCtrlProp()[utils.ENV_CTRL_CADATA]
+							if caCert == expectedCaCert {
+								portFound = true
+								break
+							} else {
+								utils.AviLog.Infof("ManagementService CA certificate mismatch: expected length %d, got length %d", len(expectedCaCert), len(caCert))
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	if !portFound {
+		utils.AviLog.Infof("ManagementService expected port configuration not found")
+		return false
+	}
+
```

Comment:
in this function, wherever we are returning false do we need Infof log or Warnf?